core/auth/src/main/java/software/amazon/awssdk/auth/signer/internal/BaseEventStreamAsyncAws4Signer.java

```diff
@@ -160,6 +161,13 @@ public ByteBuffer apply(ByteBuffer byteBuffer) {
                  * Encode signed event to byte
                  */
                 Message signedMessage = new Message(sortHeaders(headers), payload);
+
+                if (LOG.isLoggingLevelEnabled("trace")) {
```

Comment:
Log4j2 docs suggest we could use logger.isTraceEnabled() here? Might be slightly better than passing a string

Reply:
another approach is to use LOG.printf(.... which allows you to dynamically set the log level which means that theoretically you could do all this in a single line without having an if statement.

Reply:
This is our [`software.amazon.awssdk.utils.Logger`](https://github.com/aws/aws-sdk-java-v2/blob/b349c1ba93584c6489f717ef2ba7c2998a6936d4/utils/src/main/java/software/amazon/awssdk/utils/Logger.java#L25) not Log4j's

Reply:
Ahh ok nevermind then. I assume we don't have a similar method then?

Reply:
Not currently